cmd/workspace/apps/logs.go

```diff
@@ -0,0 +1,262 @@
+  databricks apps logs my-app --follow --search ERROR --source APP
+
+  # Mirror streamed logs to a local file while following for up to 5 minutes
+  databricks apps logs my-app --follow --timeout 5m --output-file /tmp/my-app.log`,
+		Args:    root.ExactArgs(1),
+		PreRunE: root.MustWorkspaceClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			if tailLines < 0 {
+				return errors.New("--tail-lines cannot be negative")
+			}
+
+			if follow && streamTimeout > 0 {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, streamTimeout)
+				defer cancel()
+			}
+
+			name := args[0]
+			w := cmdctx.WorkspaceClient(ctx)
+			app, err := w.Apps.Get(ctx, apps.GetAppRequest{Name: name})
+			if err != nil {
+				return err
+			}
+			if app.Url == "" {
+				return fmt.Errorf("app %s does not have a public URL; deploy and start it before streaming logs", name)
+			}
+
+			wsURL, err := buildLogsURL(app.Url)
+			if err != nil {
+				return err
+			}
+
+			cfg := cmdctx.ConfigUsed(ctx)
+			if cfg == nil {
+				return errors.New("missing workspace configuration")
+			}
+			authArgs := &auth.AuthArguments{Host: cfg.Host, AccountID: cfg.AccountID}
+			tokenRequest := auth.AcquireTokenRequest{
+				AuthArguments: authArgs,
+				ProfileName:   cfg.Profile,
+				Timeout:       tokenAcquireTimeout,
+			}
+			token, err := auth.AcquireToken(ctx, tokenRequest)
+			if err != nil {
+				return err
+			}
+			tokenProvider := func(ctx context.Context) (string, error) {
+				tok, err := auth.AcquireToken(ctx, tokenRequest)
+				if err != nil {
+					return "", err
+				}
+				return tok.AccessToken, nil
+			}
+
+			var writer io.Writer = cmd.OutOrStdout()
+			var file *os.File
+			if outputPath != "" {
+				file, err = os.OpenFile(outputPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o600)
+				if err != nil {
+					return err
+				}
+				defer file.Close()
+				writer = io.MultiWriter(writer, file)
+			}
+			colorizeLogs := outputPath == "" && cmdio.IsTTY(cmd.OutOrStdout())
+
+			sourceMap, err := buildSourceFilter(sourceFilters)
+			if err != nil {
+				return err
+			}
+
+			log.Infof(ctx, "Streaming logs for %s (%s)", name, wsURL)
+			return logstream.Run(ctx, logstream.Config{
+				Dialer:        newLogStreamDialer(cfg),
+				URL:           wsURL,
+				Origin:        normalizeOrigin(app.Url),
+				Token:         token.AccessToken,
+				TokenProvider: tokenProvider,
+				Search:        searchTerm,
+				Sources:       sourceMap,
+				Tail:          tailLines,
+				Follow:        follow,
+				Prefetch:      defaultPrefetchWindow,
+				Writer:        writer,
+				UserAgent:     "databricks-cli apps logs",
+				Colorize:      colorizeLogs,
+			})
+		},
+	}
+
+	streamGroup := cmdgroup.NewFlagGroup("Streaming")
+	streamGroup.FlagSet().IntVar(&tailLines, "tail-lines", defaultTailLines, "Number of recent log lines to show before streaming. Set to 0 to show everything.")
```

Comment:
I tested the flag and seems to not work when running together with `-f` flag - it always shows me all the logs instead of last n lines 🤔 It works properly though when I don't have the `follow` flag specified.